bioinformatics_utils/bioinformatics_utils.py

```diff
@@ -0,0 +1,120 @@
+from typing import List, Union, Dict, Tuple
+
+from scripts.script_dna_rna_tools import is_valid_sequence, transcribe, reverse
+from scripts.script_dna_rna_tools import complement, reverse_complement
+from scripts.script_fastq_filter import gc_content, average_quality
+from scripts.script_fastq_filter import read_fastq, write_fastq
+
+
+def run_dna_rna_tools(
+    *args: Union[str, List[str]]
+) -> Union[str, List[str], None]:
+
+    """
+    Executes specified procedures (transcription,
+    reverse, complementarity) for given
+    DNA/RNA sequences.
+
+    Args:
+        *args: DNA/RNA sequences and
+        the procedure to perform.
+               The last argument must be a string
+               indicating the procedure.
+
+    Returns:
+        The result of the procedure execution:
+        - If the procedure is "transcribe", returns
+          the transcribed sequence.
+        - If the procedure is "reverse",
+          returns the reversed sequence.
+        - If the procedure is "complement",
+          returns the complementary sequence.
+        - If the procedure is "reverse_complement",
+          returns the reverse complementary sequence.
+
+        If there is one sequence, returns
+        a string; if more than one, returns a list of strings.
+        If there are no sequences, returns
+        None.
+
+    Raises:
+        ValueError: If the sequence
+        is invalid or the procedure is unknown.
+    """
+
+    if not args:
+        return None
+
+    procedure = args[-1]
+    sequences = args[:-1]
+
+    for seq in sequences:
+        if not is_valid_sequence(seq):
+            raise ValueError(f"Invalid sequence: {seq}")
+
+    if procedure == "transcribe":
+        result = [transcribe(seq) for seq in sequences]
+    elif procedure == "reverse":
+        result = [reverse(seq) for seq in sequences]
+    elif procedure == "complement":
+        result = [complement(seq) for seq in sequences]
+    elif procedure == "reverse_complement":
+        result = [reverse_complement(seq) for seq in sequences]
+    else:
+        raise ValueError(f"Unknown procedure: {procedure}")
```

Comment:
Для более компактной записи можно как вариант использовать словарь
```suggestion
     procedures: Dict[str, Callable[[str], str]] = {
        "transcribe": transcribe,
        "reverse": reverse,
        "complement": complement,
        "reverse_complement": reverse_complement
    }

    if procedure not in procedures:
        raise ValueError(f"Unknown procedure: {procedure}")
        
    result = [procedures[procedure](seq) for seq in sequences]
```

Reply:
Не обязательно (Антону например больше нравится через услования) но, да, так можно. Как минимум полезно знать про разные варианты